data-plane/receiver-loom/src/test/java/dev/knative/eventing/kafka/broker/receiverloom/LoomKafkaProducerTest.java

```diff
@@ -0,0 +1,226 @@
+        Future<RecordMetadata> sendFuture = producer.send(record);
+
+        // Verify that the sendFuture fails with the expected error message
+        sendFuture.onComplete(ar -> {
+            testContext.verify(() -> {
+                assertTrue(ar.failed());
+                assertEquals("Producer is closed", ar.cause().getMessage());
+                testContext.completeNow();
+            });
+        });
+    }
+
+    @Test
+    public void testCloseIsWaitingForEmptyQueue(VertxTestContext testContext) {
+        // Send a record and wait for it to be processed
+        int numRecords = 100000;
+        List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        for (int i = 0; i < numRecords; i++) {
+            ProducerRecord<String, Integer> record = new ProducerRecord<>("test", "sequence number", i);
+            sendFutures.add(producer.send(record));
+        }
+
+        // Close the producer and wait for the event queue to be empty
+        producer.close().onComplete(ar -> {
+            testContext.verify(() -> {
+                assertTrue(ar.succeeded());
+
+                // Ensure that the eventQueue is empty after closing
+                assertTrue(producer.getEventQueueSize() == 0);
+                // Ensure that the sendFromQueueThread is not alive after closing
+                assertTrue(!producer.isSendFromQueueThreadAlive());
+                testContext.completeNow();
+            });
+        });
+
+        // Verify that the sent record is processed before the producer is closed
+        for (int i = 0; i < numRecords; i++) {
+            sendFutures.get(i).onComplete(ar -> {
+                testContext.verify(() -> {
+                    assertTrue(ar.succeeded());
+                    testContext.completeNow();
+                });
+            });
+        }
+    }
+
+    @Test
+    void interruptHappensWhenClose(VertxTestContext testContext) throws InterruptedException {
+        // send a single record
+        ProducerRecord<String, Integer> record = new ProducerRecord<>("test", "sequence number", 123);
+        Future<RecordMetadata> sendFuture = producer.send(record);
+
+        // wait sendFuture is completed
+        sendFuture.onComplete(ar -> {
+            testContext.verify(() -> {
+                assertTrue(ar.succeeded());
+                testContext.completeNow();
+            });
+        });
+
+        // now sendFromQueueThread should be blocked in eventQueue.take()
+
+        // close the producer
+        Future<Void> future = producer.close();
+
+        while (future.isComplete()) {
+            Thread.sleep(100);
+        }
+        // wait for 5 seconds to ensure the sendFromQueueThread is interrupted
+        Thread.sleep(5000L);
+        // verify the sendFromQueueThread is interrupted
+        testContext.verify(() -> {
+            assertTrue(!producer.isSendFromQueueThreadAlive());
+            testContext.completeNow();
+        });
```

Comment:
Hm, interesting this test is passing locally. Should I increase the wait time? Maybe it need some more time to get interrupted as it was blocked.